### PR TITLE
Client/options page

### LINKED ISSUE #45: Create options page and redirect from navbar

### LINKED ISSUE #46: Make cards of blacklisted sites; buttons to add sites - delete functionality TBD

### LINKED ISSUE #48: Options to change default allotment time (and for which period of the day)

### PR DESCRIPTION
**Pull request by xynyx**
Blacklist, add and delete functionality working. However there seems to be a bug where if you delete something, re-add it, delete it again, and then try re-adding it, the state doesn't update. If you refresh, it will appear (so probably not DB-related), but React will not auto-update it.

### CODE REVIEW

**Comment by xynyx** [comment]
Another bug - if you add a site to an empty list, it crashes. 

"TypeError: state.blacklisted is not iterable".

Haven't looked into it yet but probably an easier fix.

**Comment by xynyx** [comment]
Sometimes when you delete multiple entries that are not initially in the database quickly, they'll reappear after a refresh. Something wrong with that part of the route/dispatch

**Comment by xynyx** [comment]
UPDATE: Fixed multiple delete entry bug as well as adding an entry to an empty list. Only the first issue remains. Will not work on it any further for now.

**Review by jpqy — APPROVED** [review]
Options page UI looks great. Minor bugs that need to be fixed but I think it's best to merge now to make it available on master.

**Comment by riztaha** [comment]
Good work xynyx! UI is nice & simple, easy to add/delete and change quota. Recommend to increase cap to more than 2hrs (could be 3hrs as we discussed) or let users decide what their cap is (eg. a built-in app on my phone lets me choose app screen time freely in increments 1hrs:5mins).

Tested for add/remove website - Changes flag in DB
Did not test for Quota DB change.

Minor bug - site crashes upon deleting all websites and then trying to add a website.

Agree with jpqy, let's merge and fix this bug later.

File: server/helpers/dbHelper.js
Comment (riztaha) [inline server/helpers/dbHelper.js:423]: Minor Bug mentioned earlier for the site crash could be fixed by jpqy's suggestion of "instead of returning null, return res.rows. Null is not iterable, but an empty array is iterable"

File: server/helpers/dbHelper.js
Comment (riztaha) [inline server/helpers/dbHelper.js:439]: May need to change this return statement as well

**Review by riztaha — DISMISSED** [review]
Left inline comment for dbHelper.js. Good to merge after.

File: server/helpers/dbHelper.js
Comment (jpqy) [reply server/helpers/dbHelper.js:423]: These are fixed in my branch so no need to change. A lot more bugs needed to be fixed than just these 2.

**Comment by jpqy** [comment]
> Another bug - if you add a site to an empty list, it crashes.
> 
> "TypeError: state.blacklisted is not iterable".
> 
> Haven't looked into it yet but probably an easier fix.

> UPDATE: Fixed multiple delete entry bug as well as adding an entry to an empty list. Only the first issue remains. Will not work on it any further for now.

This bug is caused by the db query returning and the server sending back `null` for when a person has nothing in their blacklist:
```
if (res.rows.length === 0) return null;
        return res.rows;
```
The TypeError occurs because you can't iterate over null.

The fix for this is to have the server send an empty array instead of null when there is nothing in the blacklist. The good news is that `res.rows` is actually an empty array if it can't find anything. So, we can just replace the above with just:
```
return res.rows;
```

I have fixed these kind of bugs in #58. 
@xynyx: Once it is merged you can get rid of the `if` statements that were added to fix your "not iterable" bugs.

**Review by xynyx — COMMENTED** [review]
Integrated with jpqy's chrome extension branch & master; fixed all bugs and integration issues.

**Pull request merged**
